src/components/learner-credit-management/tests/BudgetDetailPage.test.jsx

```diff
@@ -922,20 +924,22 @@ describe('<BudgetDetailPage />', () => {
     if (field === 'status') {
       const filtersButton = getButtonElement('Filters', { screenOverride: assignedSection });
       userEvent.click(filtersButton);
-      const filtersDropdown = screen.getByRole('group', { name: 'Status' });
-      const filtersDropdownContainer = within(filtersDropdown);
-      if (value.includes('waiting')) {
-        const waitingForLearnerOption = filtersDropdownContainer.getByLabelText('Waiting for learner 1', { exact: false });
-        expect(waitingForLearnerOption).toBeInTheDocument();
-        userEvent.click(waitingForLearnerOption);
-
-        await waitFor(() => {
-          expect(waitingForLearnerOption).toBeChecked();
-          expect(mockFetchContentAssignments).toHaveBeenCalledWith(
-            expect.objectContaining(expectedTableFetchDataArgsAfterFilter),
-          );
-        });
-      }
+      act(async () => {
+        const filtersDropdown = await screen.findByRole('group', { name: 'Status' });
+        const filtersDropdownContainer = within(filtersDropdown);
+        if (value.includes('waiting')) {
+          const waitingForLearnerOption = filtersDropdownContainer.getByLabelText('Waiting for learner 1', { exact: false });
+          expect(waitingForLearnerOption).toBeInTheDocument();
+          userEvent.click(waitingForLearnerOption);
```

Comment:
`userEvent.click`  should not be used within an `act`. See https://kentcdodds.com/blog/common-mistakes-with-react-testing-library#wrapping-things-in-act-unnecessarily for more details.

Reply:
Since tests were failing without it `act` was used initially to ensure that all updates to the component (like state changes and DOM updates) were flushed and applied before assertions without these test was failing. However you are right, I have replaced it with `waitFor` which automatically handles the flushing of updates within its callback, waiting for the expected condition to be true which seems to have resolved the issue.